packages/app/src/app/pages/Sandbox/Editor/Workspace/Chat/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { sortBy, takeRight } from 'lodash-es';
+
+import AutosizeTextArea from '@codesandbox/common/lib/components/AutosizeTextArea';
+import { ENTER } from '@codesandbox/common/lib/utils/keycodes';
+import { useOvermind } from 'app/overmind';
+
+const Container = styled.div`
+  min-height: 200px;
+  max-height: 300px;
+  padding: 0 1rem;
+  color: white;
+  font-size: 0.875rem;
+  display: flex;
+  flex-direction: column;
+  overflow-y: auto;
+`;
+
+const Messages = styled.div`
+  height: 100%;
+  flex: 1;
+`;
+
+export const Chat: React.FC = () => {
+  const [value, setValue] = useState('');
+  const [height, setHeight] = useState('');
+  const { state, actions } = useOvermind();
+  const messagesRef = useRef(null);
+  function scrollDown() {
+    if (messagesRef.current) {
+      messagesRef.current.scrollTop = messagesRef.current.scrollHeight;
+    }
+  }
+  useEffect(scrollDown);
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e.keyCode === ENTER && !e.shiftKey) {
+      e.preventDefault();
+      e.stopPropagation();
+      // Enter
+      actions.live.onSendChat({
+        message: value,
+      });
+      setValue('');
+      scrollDown();
+    }
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const { messages, users } = state.live.roomInfo.chat;
+  const currentUserId = state.live.liveUserId;
+  const roomInfoUsers = state.live.roomInfo.users;
+
+  return (
+    <Container
+      ref={el => {
+        messagesRef.current = el;
+      }}
+    >
+      <Messages>
+        {messages.length > 0 ? (
+          sortBy(takeRight(messages, 100), 'date').map((message, i) => {
+            const metadata = roomInfoUsers.find(u => u.id === message.userId);
+            const color = metadata
+              ? `rgb(${metadata.color[0]}, ${metadata.color[1]}, ${
+                  metadata.color[2]
+                })`
+              : '#636363';
+            const name = users.get
+              ? users.get(message.userId)
```

Comment:
I did my best to extend the user type to accept a function, but I still get an error here - Typescript sees `.get` as a string:

```ts
const users: {
    [x: string]: string | ChatUserGetter;
    get: string;
}
```

 I'm lost as to why 🤔

Reply:
You can actually just remove this. This was related to previous mobx solution. So basically:

```ts
const name = users[message.userId] 
```

Is enough :)